docs/plugins.md

```diff
@@ -918,6 +918,51 @@ In the same manner additional services from webdriverio can be installed, enable
 
 -   `config`  
 
+## reRunFailedTest
+It stores failed scripts from current execution in failedCases.json
+
+This plugin allows running
+- only the scripts which are failed in previous execution
+- run any custom scripts provided by user without any pattern (can be provided in failedCases.json)
+- autoRetry failed scripts from current execution to detect flakiness
+
+```js
+plugins: {
+  reRunFailedTest: {}
+  enabled: true,
+    autoRetry: true
+}
+```
+
+Run test with plugin enabled
+```js
+npx codeceptjs run --plugins reRunFailedTest
+```
+
+#### Configuration
+- autoRetry: to auto retry the failed scripts from current execution after all scripts are completed
+
+### Options
+
+
+| Param            | Description                                                                    |
+| ---------------- | ------------------------------------------------------------------------------ |
+| failed           | Only executes the failed/custom(selective) scripts present in failedCases.json              |
+
+```js
+npx codeceptjs run --failed
```

Comment:
you should be able to read the Plugin values from the config `plugins` object. You might not need to provide CLI option unless it's a core codeceptjs feature.

Reply:
@gkushang, This option is more over to run the failed tests from the previous run to support both runner options sequential and workers. this would be a core feature as many of the developers were looking for this feature similar option in testng.

Reply:
@senthillkumar The reason I asked because I see you also have a plugin `reRunFailedTest.js` defined so I was wondering the use case of the Plugin. If this becomes the Core feature then you wont' need a plugin or vice-versa (if plugin is enabled then it'd run failed tests automatically)

** This is indeed a much needed feature though.

Reply:
> Would the user need to execute one more command to run `failed` tests or the `failed` tests will run automatically if plugin is enabled? e.g.
> 
> ` codeceptjs run` >> run all tests and re-run failed tests automatically at the end of execution
> OR
> `codeceptjs run && codeceptjs run -p <this-plugin>` >> execute two commands?

There are 2 features in the rerun test plugin.
1. Manually run the failed test cases using run/run-workers command with --failed option as mentioned below.
 `codeceptjs run --failed or codeceptjs run-workers --failed`
2. Enabling the Rerunfailed test Plugin would auto retry the failed test after the execution without passing any argument.

>     reRunFailedTest: {
>       enabled: true,
>       autoRetry: true,
>       require: './plugins/reRunFailedTest',
>     },

Advantages:
As i have 50 test cases running in parallel out of 50 , 10 were failed. i want to autoretry / manual retry the failed test case in the same/different environment. As rerun functionality is available as core feature in the codeceptjs, but that will rerun the scripts sequentially and if the case is failed in between it will retry the failed one then continue the rest of the scripts. rerun mode is more time consuming. 

Let me know your thoughts.

Reply:
@gkushang , can you please review the changes